packages/pools/src/concentrated/fetch-tick-data-provider.ts

```diff
@@ -0,0 +1,185 @@
+import { Dec, Int } from "@keplr-wallet/unit";
+import {
+  estimateInitialTickBound,
+  LiquidityDepth,
+  maxTick,
+  minTick,
+} from "@osmosis-labs/math";
+
+import {
+  ConcentratedLiquidityPool,
+  TickDataProvider,
+  TickDepths,
+} from "./pool";
+
+type TickDepthsResponse = {
+  liquidity_depths: {
+    liquidity_net: string;
+    tick_index: string;
+  }[];
+};
+
+/** Default tick data provider that fetches ticks for a single CL pool with `fetch` if the environment supports it, if not a fetcher can be supplied. */
+export class FetchTickDataProvider implements TickDataProvider {
+  protected _zeroForOneTicks: LiquidityDepth[] = [];
+  protected _oneForZeroTicks: LiquidityDepth[] = [];
+
+  protected _zeroForOneBoundIndex = new Int(0);
+  protected _oneForZeroBoundIndex = new Int(0);
+
+  constructor(
+    protected readonly baseNodeUrl: string,
+    protected readonly poolId: string,
+    protected readonly nextTicksRampMultiplier = new Int(9),
+    protected readonly maxNumRequeriesPerDenom = 9,
+    protected readonly tickFetcher: (
+      poolId: string,
+      tokenInDenom: string,
+      boundTickIndex: string
+    ) => Promise<TickDepthsResponse> = async (
+      poolId,
+      tokenInDenom,
+      boundTickIndex
+    ) => {
+      if (fetch === undefined)
+        throw new Error("Fetch method must be available in the environment");
+
+      const res = await fetch(
+        `${this.baseNodeUrl}osmosis/concentratedliquidity/v1beta1/liquidity_net_in_direction?pool_id=${poolId}&token_in=${tokenInDenom}&use_cur_tick=true&bound_tick=${boundTickIndex}`
```

Comment:
Consider using `new URL`, and `url.searchParams.append` here

Reply:
Will have GPT4 do this

Reply:
https://chat.openai.com/share/3790441c-57e1-49bf-9fae-085447fd3825

It was great at this